### PR TITLE
IOS integration part 1

### PR DESCRIPTION
**Pull request by 3nids**
app works in simulator (demo projects, proj and QGIS SVGs are ok)
GPS and camera not properly handled yet

### CODE REVIEW

File: src/core/qgismobileapp.h [hunk 1]
```diff
@@ -43,6 +43,8 @@
 #include "platformutilities.h"
 #if defined(Q_OS_ANDROID)
 #include "androidplatformutilities.h"
+#elif defined(Q_OS_IOS)
+#include "ios/iosplatformutilities.h"
```
Comment (m-kuhn) [inline src/core/qgismobileapp.h:47]: Is this needed?

File: src/core/CMakeLists.txt [hunk 2]
```diff
@@ -167,21 +167,28 @@ IF (ANDROID)
...
+  )
+  SET(QFIELD_CORE_SRCS ${QFIELD_CORE_SRCS}
+    ios/iosplatformutilities.mm
+  )
 ENDIF (ANDROID)
 
 FIND_PACKAGE(Sqlite3)
 
-ADD_LIBRARY(qfield_core SHARED ${QFIELD_CORE_SRCS} ${QFIELD_CORE_HDRS})
+ADD_LIBRARY(qfield_core STATIC ${QFIELD_CORE_SRCS} ${QFIELD_CORE_HDRS})
```
Comment (m-kuhn) [inline src/core/CMakeLists.txt:181]: Always static?

**Review by m-kuhn — APPROVED** [review]
:green_apple:

File: src/core/CMakeLists.txt (hunk 2 above)
Comment (3nids) [reply src/core/CMakeLists.txt:181]: that's what @OlivierLDff wrote. is this an issue?

Comment (m-kuhn) [reply src/core/CMakeLists.txt:181]: Probably not. And if it is, then we can switch again.

Comment (OlivierLDff) [reply src/core/CMakeLists.txt:181]: I would say, always use static by default:
- This remove one step for deployment
- Only the required symbols are embedded making the app smaller

I feel we are in era of packaging everything and creating separate environment for every application.

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 76c5c2bc798a44776783e1a4eeaa64d8e3da0fcb:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1769-1497068485e38db3170ac576100bb021f19e1b38-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1769-1497068485e38db3170ac576100bb021f19e1b38-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1769-1497068485e38db3170ac576100bb021f19e1b38-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1769-1497068485e38db3170ac576100bb021f19e1b38-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 1a203a81145b7c9922dc7d61247e84ca265b7c8b:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1769-732a7b4649e5d090273635e7db08293d0af67c17-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1769-732a7b4649e5d090273635e7db08293d0af67c17-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1769-732a7b4649e5d090273635e7db08293d0af67c17-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1769-732a7b4649e5d090273635e7db08293d0af67c17-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for fcf7f69e0be6a5d7d1070ee0a98e57551a85eadb:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1769-b2ef6c1f3d59b1956c4c888ac7f7094a0e7334b0-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1769-b2ef6c1f3d59b1956c4c888ac7f7094a0e7334b0-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1769-b2ef6c1f3d59b1956c4c888ac7f7094a0e7334b0-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1769-b2ef6c1f3d59b1956c4c888ac7f7094a0e7334b0-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for a7c50ce99f24824670970a0329c2072bcfe47b33:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1769-61c391b6054c6c39c867e8e1677c0cf243b6e680-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1769-61c391b6054c6c39c867e8e1677c0cf243b6e680-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1769-61c391b6054c6c39c867e8e1677c0cf243b6e680-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1769-61c391b6054c6c39c867e8e1677c0cf243b6e680-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for cd17f56840c036b5c6e7d5a2c0197daf89875b26:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1769-06224393fcb0a8dd15eca345419003a46f83b20a-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1769-06224393fcb0a8dd15eca345419003a46f83b20a-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1769-06224393fcb0a8dd15eca345419003a46f83b20a-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1769-06224393fcb0a8dd15eca345419003a46f83b20a-x86.apk)

**Pull request merged**